src/util/run.cpp

```diff
@@ -101,6 +102,18 @@ int run(
     }
   }
 
+  int stderr_fd = STDERR_FILENO;
+
+  if(!std_error.empty())
+  {
+    stderr_fd = open(std_error.c_str(), O_CREAT | O_WRONLY, S_IRUSR | S_IWUSR);
+    if(stderr_fd == -1)
+    {
+      perror("Failed to open stderr copy");
+      return 1;
+    }
+  }
```

Comment:
Wishlist item: we are doing almost the same three times now, I'd suggest:
````c++
static int stdio_copy(int fd, const std::string &file)
{
  int result_fd = fd;

  if(file.empty())
    return result_fd;

  int flags = 0, mode = 0;
  std::string name;

  switch(fd)
  {
    case STDIN_FILENO:
      flags = O_RDONLY;
      name = "stdin";
   case STDOUT_FILENO:
   case STDERR_FILENO:
     flags = O_CREAT | O_WRONLY;
     mode = S_IRUSR | S_IWUSR;
     name = fd == STDOUT_FILENO ? "stdout" : "stderr";
    default:
      UNREACHABLE;
  }

  result_fd = open(file.c_str(), flags, mode);
  if(result_fd == -1)
    perror("Failed to open " + name + " copy");

  return result_fd;
}
````

Reply:
ok, done